### PR TITLE
disintegration

### PR DESCRIPTION
**Pull request by turbocrime**
makes interfaces reasonable

removes services-context package, querier package, storage package

creates block processor package, idb package

most of this diff is changing tests, which for some reason were heavily dependent on interfaces

### CODE REVIEW

**Review by VanishMax — APPROVED** [review]
Major changes here, looking all good to me! It's an approve from me but I believe it would be beneficial for other team members to look at it too.

File: packages/storage/src/version.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export default 42;
```
Comment (TalDerei) [inline packages/storage/src/version.ts]: can we export a variable, something like

```
/**
 * The version number for the IndexedDB schema. This version number is used to manage
 * database upgrades and ensure that the correct schema version is applied.
 */
export const IDB_VERSION = 42;
```

**Comment by TalDerei** [comment]
remind me why we removed the querier in the block processor?

**Comment by grod220** [comment]
I'll take a look at this more later in the afternoon, but after looking through 500+ lines of code, this does not feel like a reasonable chunk of code we can ship at once. There are meaningful changes to logic and code organization that we need to break down into multiple PRs.

**Pull request closed without merging**